config.ru

```diff
@@ -11,6 +11,19 @@ map '/lobster' do
   run Rack::Lobster.new
 end
 
+map '/headers' do
+  headers = proc do |env|
+    [200, { "Content-Type" => "text/html" }, [
+      env.select {|key,val| key.start_with? 'HTTP_'}
+      .collect {|key, val| [key.sub(/^HTTP_/, ''), val]}
```

Comment:
The original headers don't have `HTTP_` prefix. AFAIK it is rack that puts them into `env` by adding the prefix. So `HTTP_` is not actually part of the header name. See http://www.rubydoc.info/github/rack/rack/file/SPEC